src/PerformanceSensitiveAnalyzers/CSharp/CodeFixes/AvoidAllocationWithArrayEmptyCodeFix.cs

```diff
@@ -0,0 +1,283 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.PerformanceSensitiveAnalyzers;
+
+namespace Microsoft.CodeAnalysis.CSharp.PerformanceSensitiveAnalyzers.CodeFixes
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(AvoidAllocationWithArrayEmptyCodeFix)), Shared]
+    internal sealed class AvoidAllocationWithArrayEmptyCodeFix : CodeFixProvider
+    {
+        private readonly string _title = CodeFixesResources.AvoidAllocationByUsingArrayEmpty;
+
+        public sealed override ImmutableArray<string> FixableDiagnosticIds
+            => ImmutableArray.Create(ExplicitAllocationAnalyzer.ObjectCreationRuleId, ExplicitAllocationAnalyzer.ArrayCreationRuleId);
+
+        public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var diagnostic = context.Diagnostics.First();
+            var diagnosticSpan = diagnostic.Location.SourceSpan;
+            var node = root.FindNode(diagnosticSpan);
+
+            if (IsReturnStatement(node))
+            {
+                await TryToRegisterCodeFixesForReturnStatement(context, node, diagnostic).ConfigureAwait(false);
+                return;
+            }
+
+            if (IsMethodInvocationParameter(node))
+            {
+                await TryToRegisterCodeFixesForMethodInvocationParameter(context, node, diagnostic).ConfigureAwait(false);
+                return;
+            }
+        }
+
+        private async Task TryToRegisterCodeFixesForMethodInvocationParameter(CodeFixContext context, SyntaxNode node, Diagnostic diagnostic)
+        {
+            var semanticModel = await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+            if (IsExpectedParameterReadonlySequence(node, semanticModel) && node is ArgumentSyntax argument)
+            {
+                TryRegisterCodeFix(context, node, diagnostic, argument.Expression, semanticModel);
+            }
+        }
+
+        private async Task TryToRegisterCodeFixesForReturnStatement(CodeFixContext context, SyntaxNode node, Diagnostic diagnostic)
+        {
+            var semanticModel = await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+
```

Comment:
[cosmetic] Either remove this blank line or add one between line 50 and 51 for consistency.